packages/components/src/RBACProvider/RBACProvider.tsx

```diff
@@ -0,0 +1,62 @@
+import React, { useEffect, useRef, useState } from 'react';
+import { Access } from '@redhat-cloud-services/rbac-client';
+import { Bullseye, Spinner } from '@patternfly/react-core';
+
+import {
+  RBACContext,
+  doesHavePermissions,
+  getRBAC,
+  hasAllPermissions,
+  initialPermissions,
+} from '@redhat-cloud-services/frontend-components-utilities/RBAC';
+
+const hasAccessWithUserPermissions = (userPermissions: (Access | string)[]) => {
+  return (requiredPermissions: string[], checkAll = true): boolean => {
+    return checkAll ? hasAllPermissions(userPermissions, requiredPermissions) : doesHavePermissions(userPermissions, requiredPermissions);
+  };
+};
+
+export interface RBACProviderProps {
+  appName: string;
+}
+
+export const RBACProvider: React.FunctionComponent<RBACProviderProps> = ({ appName, children }) => {
+  const loading = useRef(false);
```

Comment:
Any reason for using ref instead of state? The ref change will not trigger a re-render, meaning when app initiates a re-fetch (when for instance `appName` changes) the UI will not show spinner.

Reply:
I think we actually don't need this anymore. This was to prevent the `state update on unmounted component error`. I'll try to actually remove it and see how it behaves without it.